sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -899,6 +919,21 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
           spec, partitionColumnNames, tablePath)
         try {
           tablePath.getFileSystem(hadoopConf).rename(wrongPath, rightPath)
+
+          // If the newSpec contains more than one depth partition, FileSystem.rename just deletes
+          // the leaf(i.e. wrongPath), we should check if wrongPath's parents need to be deleted.
+          // For example, give a newSpec 'A=1/B=2', after calling Hive's client.renamePartitions,
+          // the location path in FileSystem is changed to 'a=1/b=2', which is wrongPath, then
+          // although we renamed it to 'A=1/B=2', 'a=1/b=2' in FileSystem is deleted, but 'a=1'
+          // is still exists, which we also need to delete
+          val delHivePartPathAfterRename = getExtraPartPathCreatedByHive(
```

Comment:
So far, the partition rename DDL we support is for a single pair of partition spec. That is, `ALTER TABLE table PARTITION spec1 RENAME TO PARTITION spec2`. This is not an issue for end users.

Thus, your concern looks reasonable, but I think we should not support the multi-partition renaming in the SessionCatalog and ExternalCatalog. It just makes the code more complex for error handling. Let me remove it.